### PR TITLE
Update devfile integration tests to OpenShift 4.15

### PR DESCRIPTION
**Pull request by michael-valdron**
# Description

Devfile integration tests will be using golang 1.21 and kubernetes 1.29 after pending changes: https://github.com/devfile/integration-tests/pull/105

blocking devfile/api#1561

### CODE REVIEW

**Comment by thepetk** [comment]
/pj-rehearse periodic-ci-devfile-integration-tests-main-v4.15.console-e2e-aws-console-periodic

**Comment by thepetk** [comment]
/pj-rehearse pull-ci-devfile-integration-tests-main-v4.15.console-images

**Comment by michael-valdron** [comment]
/pj-rehearse ack

Existing console checks are failing as well for the past year, given that there is no solution to resolve these at this time though intent on resolving it at a future date we can proceed with adding the OCP 4.15 jobs despite the failing console jobs that we intent to be resolved with the existing ones.

**Review by johnmcollier — COMMENTED** [review]
Why aren't we migrating one of the existing job configs? I don't think we need to keep both 4.10 or 4.12 around

**Comment by michael-valdron** [comment]
> Why aren't we migrating one of the existing job configs? I don't think we need to keep both 4.10 or 4.12 around

@johnmcollier we could scrap the oldest 4.10 as I believe its EOL by now.

**Comment by johnmcollier** [comment]
Removing 4.10 would be fine

File: ci-operator/config/devfile/integration-tests/devfile-integration-tests-main__v4.15.console.yaml [hunk 1]
```diff
@@ -40,7 +40,7 @@ resources:
       memory: 8Gi
 tests:
 - as: e2e-aws-console-periodic
-  cron: 12 0 3,19 * *
+  cron: 0 2 * * 1-5
```
Comment (johnmcollier) [inline ci-operator/config/devfile/integration-tests/devfile-integration-tests-main__v4.15.console.yaml:43]: Use the old cron configuration for this job. It'll reduce how often the (currently) broken tests run

File: ci-operator/jobs/devfile/integration-tests/devfile-integration-tests-main-periodics.yaml [hunk 2]
```diff
@@ -12,10 +12,10 @@ periodics:
   labels:
     ci-operator.openshift.io/cloud: aws
     ci-operator.openshift.io/cloud-cluster-profile: aws-devfile
-    ci-operator.openshift.io/variant: v4.10.console
+    ci-operator.openshift.io/variant: v4.11.console
```
Comment (johnmcollier) [inline ci-operator/jobs/devfile/integration-tests/devfile-integration-tests-main-periodics.yaml:15]: v4.11?

File: ci-operator/jobs/devfile/integration-tests/devfile-integration-tests-main-periodics.yaml [hunk 3]
```diff
@@ -12,10 +12,10 @@ periodics:
...
     ci.openshift.io/generator: prowgen
     pj-rehearse.openshift.io/can-be-rehearsed: "true"
-  name: periodic-ci-devfile-integration-tests-main-v4.10.console-e2e-aws-console-periodic
+  name: periodic-ci-devfile-integration-tests-main-v4.11.console-e2e-aws-console-periodic
```
Comment (johnmcollier) [inline ci-operator/jobs/devfile/integration-tests/devfile-integration-tests-main-periodics.yaml:18]: v4.11? (and same for odo down below)

File: ci-operator/jobs/devfile/integration-tests/devfile-integration-tests-main-periodics.yaml [hunk 4]
```diff
@@ -1,7 +1,7 @@
 periodics:
 - agent: kubernetes
   cluster: build05
-  cron: 12 0 3,19 * *
+  cron: 0 2 * * 5
```
Comment (johnmcollier) [inline ci-operator/jobs/devfile/integration-tests/devfile-integration-tests-main-periodics.yaml:4]: Use old cron configuration for console

File: ci-operator/jobs/devfile/integration-tests/devfile-integration-tests-main-periodics.yaml [hunk 5]
```diff
@@ -161,7 +161,7 @@ periodics:
         secretName: result-aggregator
 - agent: kubernetes
   cluster: build05
-  cron: 0 2 * * 5
+  cron: 0 2 * * 1-5
```
Comment (johnmcollier) [inline ci-operator/jobs/devfile/integration-tests/devfile-integration-tests-main-periodics.yaml:164]: Use old cron configuration

**Review by johnmcollier — COMMENTED** [review]
Left some additional comments. I'm also seeing a lot of references to v4.11 get added in

File: ci-operator/jobs/devfile/integration-tests/devfile-integration-tests-main-periodics.yaml (hunk 2 above)
Comment (michael-valdron) [reply ci-operator/jobs/devfile/integration-tests/devfile-integration-tests-main-periodics.yaml:15]: @johnmcollier Changes to the jobs are reordered so its changing all of them to the next one up, same result though.

File: ci-operator/config/devfile/integration-tests/devfile-integration-tests-main__v4.15.console.yaml (hunk 1 above)
Comment (michael-valdron) [reply ci-operator/config/devfile/integration-tests/devfile-integration-tests-main__v4.15.console.yaml]: fixed in https://github.com/openshift/release/pull/53140/commits/54db7b92113f19e23f7d85c21f88e565c5bc6752

File: ci-operator/jobs/devfile/integration-tests/devfile-integration-tests-main-periodics.yaml (hunk 4 above)
Comment (michael-valdron) [reply ci-operator/jobs/devfile/integration-tests/devfile-integration-tests-main-periodics.yaml:4]: fixed in https://github.com/openshift/release/pull/53140/commits/41c10053c310963005662e32116fe5af9a32637b

File: ci-operator/jobs/devfile/integration-tests/devfile-integration-tests-main-periodics.yaml (hunk 5 above)
Comment (michael-valdron) [reply ci-operator/jobs/devfile/integration-tests/devfile-integration-tests-main-periodics.yaml:164]: fixed in https://github.com/openshift/release/pull/53140/commits/41c10053c310963005662e32116fe5af9a32637b

File: ci-operator/jobs/devfile/integration-tests/devfile-integration-tests-main-periodics.yaml (hunk 3 above)
Comment (michael-valdron) [reply ci-operator/jobs/devfile/integration-tests/devfile-integration-tests-main-periodics.yaml:18]: https://github.com/openshift/release/pull/53140#discussion_r1638720416

**Review by johnmcollier — APPROVED** [review]
/lgtm
/approve

**Pull request merged**